lib/asyn-thread.c

```diff
@@ -756,6 +756,14 @@ Curl_addrinfo *Curl_resolver_getaddrinfo(struct connectdata *conn,
   hints.ai_socktype = (conn->transport == TRNSPRT_TCP)?
     SOCK_STREAM : SOCK_DGRAM;
 
+#ifdef USE_LIBIDN2
+  /* done earlier */
+#elif defined(__APPLE__)
```

Comment:
This should probably be made a configure option, shouldn't it?

Reply:
Well, if you configure to use LibIDN, we assume you want to use LibIDN instead of some OS behavior.

Reply:
I have two concerns about always enabling this on Apple operating systems: (1) some people might want to turn IDNs off (usually for security reasons), and (2) it's also a configure option on Windows, so making it an option on Apple operating systems would maintain consistency.

Reply:
Feel free to make the changes for another configure option if you like.